### ISSUE TITLE
Architecture clean-up

### ISSUE THREAD

**Issue by bookercodes** [nsebhastian/react-chatbox-component#5]
From the point of view of an end-user, things are in nice shape. I haven't tested, but I am assuming someone can `npm i react-chatbox-component` and `import` the module. 

One question,

When someone runs `npm i react-chatbox-component`, will they also be downloading the `examples` and `public` folder? Seems like it adds weight to to the module for no reason.

I still feel like the structure should be,

```
+ source
- .. index.js
- .. MessageList.js
- .. Chat.js
- .. package.json

+ examples
+ .. embedded
- .... App.js
- .... style.css
- .... package.json

```

Then, we publish `source` to npm. The `examples` folder should not be in the registry. 

If importing `../../source/index.js ` from `examples/embedded` proves to be troublesome (as I expect it might based on your comment in #2), you could  `npm install react-chatbox-component` - I don't think that would be so bad? Remember, there should be two package.json files - one for `source` and one for `examples` as they have different dependencies, versions, etc.

I think starting with Create React App tricked us, because we are not creating an _app_, we are creating a lib. Whilst `examples/embedded` is an app (you could use Create React app for the example), the main project is a library and so it's peculiar that there's a `public/index.html` file in the root. I feel like that shouldn't exist.

What do you think?

**Comment by nsebhastian** [comment]
Nope. If you look into `package.json` file, there is a `build` command that will take everything inside `src/lib` and compile them into a single `dist/index.js` file.

Then there are also `main` and `files` entry, which specifically tells npm to only pick these files into the registry. So don't worry about the production module for now 😉

I've given some thoughts to this, and I think I know the problem now.

The thinking behind this whole setup is to create a `src/lib` directory where the code for the library will be published, but then if there are other developers that want to **contribute**, they can clone the repo and hack the `examples` directory instead of, say, creating their own React files then importing the `lib`

But to you and me, we want to use this `examples` as.. well, examples 😅 this is where **end users** who want to use our library get a feel of the possibilities this library give them

**Comment by nsebhastian** [comment]
What we want to achieve is different from the setup. But if we eject from Create React App, we risk opening the flood gates of babel and webpack files into the `examples` folder.. Maybe we should just create a different repo for the examples? This is the main repo for the lib, then another one for the examples

**Comment by nsebhastian** [comment]
Then we rename `examples` into `test` because that is where contributors test the library, this whole repo isn't for end users, just contributors and curious people who like to take a peak into the code

What do you think?

**Comment by bookercodes** [comment]
I think that sounds good. If we can, I would like one GitHub repository (a monolithic repository) to keep things in the same place. Easier for readers, and easier for us to maintain.

Will ultimately leave this with you to make the best decision now !

**Comment by nsebhastian** [comment]
I see... I didn't see any way to do that unless we eject or customize each CRA to allow importing from outside of `src` for now. I think the best we can do is link to individual examples and the corresponding tutorials in the README document.

Thanks for the thoughts being put into this 👍
